pilot/pkg/xds/filters/filters.go

```diff
@@ -113,7 +113,9 @@ var (
 	TLSInspector = &listener.ListenerFilter{
 		Name: wellknown.TLSInspector,
 		ConfigType: &listener.ListenerFilter_TypedConfig{
-			TypedConfig: protoconv.MessageToAny(&tlsinspector.TlsInspector{}),
+			TypedConfig: protoconv.MessageToAny(&tlsinspector.TlsInspector{
+				InitialReadBufferSize: &wrapperspb.UInt32Value{Value: 8 * 1024}, // Default is 64KB.
```

Comment:
Linkerd does 512,do we really need 8k?

Reply:
It could probably be useful to make some tcpdumps and check whats the usual mTLS client hello in standard Istio deployments

Reply:
I definitely think we do not need 8k. Just set 8k to align with http inspector. But we can set to much lower value. How about env var with defaults to 512 and people can change if needed? If every thing fine, we can remove that after couple of releases?

Reply:
I am wondering why envoy maeke 8K as default

Reply:
Legacy reasons. I do not think there is any special reason

Reply:
I don't think we need an env var since this is just the initial size, it can always resize ?

Reply:
yes. It can resize

Reply:
8k is because of MTU. that's typical for a cloud network, so it is wasteful not to read it all in one syscall.